hazelcast-jet-core/src/main/java/com/hazelcast/jet/pipeline/SourceBuilder.java

```diff
@@ -0,0 +1,514 @@
+     * but there is no hard limit). If it doesn't have any items ready, it may
+     * also return without adding anything. Once it has added all the data, it
+     * must call {@link SourceBuffer#close() buffer.close()}.
+     * <p>
+     * Unless you call {@link SourceBuilder.Batch#distributed(int) builder.distributed()},
+     * Jet will create just a single worker that should emit all the data. If
+     * you do call it, make sure your distributed source takes care of
+     * splitting the data between workers. Your {@code createFn} should consult
+     * {@link Processor.Context#totalParallelism() procContext.totalParallelism()}
+     * and {@link Processor.Context#globalProcessorIndex() procContext.globalProcessorIndex()}.
+     * Jet calls it exactly once with each {@code globalProcessorIndex} from 0
+     * to {@code totalParallelism - 1} and each of the resulting state objects
+     * must emit its unique slice of the total source data.
+     * <p>
+     * Here's an example that builds a simple, non-distributed source that
+     * reads the lines from a single text file. Since you can't control on
+     * which member of the Jet cluster the source's worker will run, the file
+     * should be available on all members. The source emits one line per {@code
+     * fillBufferFn} call.
+     * <pre>{@code
+     * BatchSource<String> fileSource = SourceBuilder
+     *         .batch("file-source", x ->
+     *             new BufferedReader(new FileReader("input.txt")))
+     *         .<String>fillBufferFn((in, buf) -> {
+     *             String line = in.readLine();
+     *             if (line != null) {
+     *                 buf.add(line);
+     *             } else {
+     *                 buf.close();
+     *             }
+     *         })
+     *         .destroyFn(BufferedReader::close)
+     *         .build();
+     * }</pre>
+     *
+     * @param name a descriptive name for the source (for diagnostic purposes)
+     * @param createFn a function that creates the state object
+     * @param <S> type of the state object
+     */
+    public static <S> SourceBuilder<S>.Batch<Void> batch(
+            @Nonnull String name, @Nonnull DistributedFunction<? super Processor.Context, ? extends S> createFn
+    ) {
+        return new SourceBuilder<S>(name, createFn).new Batch<Void>();
+    }
+
+    /**
+     * Returns a fluent-API builder with which you can create an {@linkplain
+     * StreamSource unbounded stream source} for a Jet pipeline.
+     * <p>
+     * Each parallel worker that drives your source has its private instance of
+     * a state object it gets from your {@code createFn}. To get the data items
+     * to emit to the pipeline, the worker repeatedly calls your {@code
+     * fillBufferFn} with the state object and a buffer object. Your function
+     * should add some items to the buffer, ideally those it has ready without
+     * having to block. It shouldn't add too much data at once (64 KB is enough,
```

Comment:
what does 64kb refer to here? I think number of records rather than size would be more relevant for the user.